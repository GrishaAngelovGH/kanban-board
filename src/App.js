import { useState } from "react"

import Layout from "components/Layout"
import Header from "components/Header"
import KanbanBoard from "components/KanbanBoard"
import Settings from "components/Settings"

import boardRepository from "persistent/persistentKanbanBoardRepository"
import settingsRepository from "persistent/persistentSettingsRepository"

function App() {
  const [showSettings, setShowSettings] = useState(false)
  const [showCalendar, setShowCalendar] = useState(false)
  const [downloadJsonHref, setDownloadJsonHref] = useState("")

  const toggleSettings = () => {
    setShowSettings(!showSettings)
  }

  const toggleCalendar = () => {
    setShowCalendar(!showCalendar)
  }

  const handleKanbanBoardUpdate = () => {
    const jsonString = boardRepository.getStringifiedColumns()
    const blob = new Blob([jsonString], { type: 'application/json' })
    setDownloadJsonHref(URL.createObjectURL(blob))
  }

  const handleConfirm = settings => {
    settingsRepository.saveSettings(settings)
    setShowSettings(false)
  }

  return (
    <div className="container-fluid">
      <Layout
        header={<Header downloadJsonHref={downloadJsonHref} onSettingsClick={toggleSettings} onCalendarClick={toggleCalendar} />}
        body={<KanbanBoard showCalendar={showCalendar} onUpdate={handleKanbanBoardUpdate} />}
      />

      <Settings
        show={showSettings}
        onClose={toggleSettings}
        onConfirm={handleConfirm}
      />
    </div>
  );
}

export default App;
