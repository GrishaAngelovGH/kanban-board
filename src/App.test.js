import { render } from "@testing-library/react"

import App from "./App"

window.URL.createObjectURL = jest.fn()

afterEach(() => {
  window.URL.createObjectURL.mockReset()
})

test("should render App component", () => {
  const view = render(<App />)

  expect(view).toMatchSnapshot()
})