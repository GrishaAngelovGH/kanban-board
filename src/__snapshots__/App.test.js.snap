// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid"
      >
        <div
          class="row"
        >
          <div
            class="col-md-12"
          >
            <div
              class="row layout-header"
            >
              <div
                class="col-md-12"
              >
                <div
                  class="row bg-secondary-subtle p-2 justify-content-between align-items-center"
                >
                  <div
                    class="col-4 col-md-4 col-lg-3"
                  >
                    <h3
                      class="m-0"
                    >
                      Kanban Board
                    </h3>
                  </div>
                  <div
                    class="col-4 col-md-2 col-lg-1"
                  >
                    <div
                      class="d-flex justify-content-evenly"
                    >
                      <div
                        class="dropdown-menu-header dropdown btn-group"
                        role="group"
                      >
                        <button
                          aria-expanded="false"
                          class="dropdown-toggle btn btn-light btn-sm"
                          type="button"
                        >
                          <i
                            class="bi bi-filter"
                          />
                        </button>
                      </div>
                      <div
                        class="dropdown-menu-header dropdown btn-group"
                        role="group"
                      >
                        <button
                          aria-expanded="false"
                          class="dropdown-toggle btn btn-light btn-sm"
                          type="button"
                        >
                          <i
                            class="bi bi-grid-fill text-primary"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row layout-body"
            >
              <div
                class="col-md-12"
              >
                <div
                  class="row kanban-board"
                  style="background-image: url(geometric-triangle-shapes-background.jpg);"
                >
                  <div
                    class="col-md-12"
                  >
                    <div
                      class="toast-container top-0 start-50 translate-middle-x"
                    />
                    <div
                      class="row justify-content-center p-5 text-secondary"
                    >
                      <div
                        class="col-12 col-lg-7 text-center"
                      >
                        <i
                          class="bi bi-folder-plus"
                          style="font-size: 150px;"
                        />
                        <h1>
                          Empty Kanban Board
                        </h1>
                        <h3>
                          You could click on the auto-generate board button 
                          <i
                            class="bi bi-magic"
                          />
                           or add columns and tasks manually
                        </h3>
                      </div>
                    </div>
                    <div
                      class="row p-5"
                    />
                    <div
                      class="row border-top border-secondary-subtle fixed-bottom p-2 justify-content-end text-end button-panel"
                    >
                      <div
                        class="col-md-5 col-lg-3"
                      >
                        <div
                          class="btn-group"
                          role="group"
                        >
                          <button
                            class="bi bi-magic btn btn-primary"
                            type="button"
                          />
                          <button
                            class="btn btn-primary"
                            type="button"
                          >
                            New Column
                          </button>
                          <button
                            class="btn btn-danger"
                            type="button"
                          >
                            Clear Board
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  id="DndDescribedBy-0"
                  style="display: none;"
                >
                  
    To pick up a draggable item, press the space bar.
    While dragging, use the arrow keys to move the item.
    Press space again to drop the item in its new position, or press escape to cancel.
  
                </div>
                <div
                  aria-atomic="true"
                  aria-live="assertive"
                  id="DndLiveRegion-0"
                  role="status"
                  style="position: fixed; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%); white-space: nowrap;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <div
            class="row layout-header"
          >
            <div
              class="col-md-12"
            >
              <div
                class="row bg-secondary-subtle p-2 justify-content-between align-items-center"
              >
                <div
                  class="col-4 col-md-4 col-lg-3"
                >
                  <h3
                    class="m-0"
                  >
                    Kanban Board
                  </h3>
                </div>
                <div
                  class="col-4 col-md-2 col-lg-1"
                >
                  <div
                    class="d-flex justify-content-evenly"
                  >
                    <div
                      class="dropdown-menu-header dropdown btn-group"
                      role="group"
                    >
                      <button
                        aria-expanded="false"
                        class="dropdown-toggle btn btn-light btn-sm"
                        type="button"
                      >
                        <i
                          class="bi bi-filter"
                        />
                      </button>
                    </div>
                    <div
                      class="dropdown-menu-header dropdown btn-group"
                      role="group"
                    >
                      <button
                        aria-expanded="false"
                        class="dropdown-toggle btn btn-light btn-sm"
                        type="button"
                      >
                        <i
                          class="bi bi-grid-fill text-primary"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="row layout-body"
          >
            <div
              class="col-md-12"
            >
              <div
                class="row kanban-board"
                style="background-image: url(geometric-triangle-shapes-background.jpg);"
              >
                <div
                  class="col-md-12"
                >
                  <div
                    class="toast-container top-0 start-50 translate-middle-x"
                  />
                  <div
                    class="row justify-content-center p-5 text-secondary"
                  >
                    <div
                      class="col-12 col-lg-7 text-center"
                    >
                      <i
                        class="bi bi-folder-plus"
                        style="font-size: 150px;"
                      />
                      <h1>
                        Empty Kanban Board
                      </h1>
                      <h3>
                        You could click on the auto-generate board button 
                        <i
                          class="bi bi-magic"
                        />
                         or add columns and tasks manually
                      </h3>
                    </div>
                  </div>
                  <div
                    class="row p-5"
                  />
                  <div
                    class="row border-top border-secondary-subtle fixed-bottom p-2 justify-content-end text-end button-panel"
                  >
                    <div
                      class="col-md-5 col-lg-3"
                    >
                      <div
                        class="btn-group"
                        role="group"
                      >
                        <button
                          class="bi bi-magic btn btn-primary"
                          type="button"
                        />
                        <button
                          class="btn btn-primary"
                          type="button"
                        >
                          New Column
                        </button>
                        <button
                          class="btn btn-danger"
                          type="button"
                        >
                          Clear Board
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                id="DndDescribedBy-0"
                style="display: none;"
              >
                
    To pick up a draggable item, press the space bar.
    While dragging, use the arrow keys to move the item.
    Press space again to drop the item in its new position, or press escape to cancel.
  
              </div>
              <div
                aria-atomic="true"
                aria-live="assertive"
                id="DndLiveRegion-0"
                role="status"
                style="position: fixed; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%); white-space: nowrap;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
